video/software/src/decoder/h263.rs

```diff
@@ -70,21 +75,49 @@ impl VideoDecoder for H263Decoder {
             .ok_or(H263Error::MissingWidthHeight)?;
         let (y, b, r) = picture.as_yuv();
 
-        let mut data = y.to_vec();
-        data.extend(b);
-        data.extend(r);
-
-        Ok(DecodedFrame::new(
-            width as u32,
-            height as u32,
-            BitmapFormat::Yuv420p,
-            data,
-        ))
+        let hdr = picture.as_header();
+
+        if self.1 == VideoDeblocking::Level1
+            || (self.1 == VideoDeblocking::UseVideoPacketValue
+                && hdr.options.contains(PictureOption::USE_DEBLOCKER))
+        {
+            let chroma_width = picture.chroma_samples_per_row();
+            let quantizer = hdr.quantizer as usize;
+            let strength = QUANT_TO_STRENGTH[quantizer];
```

Comment:
I think it's *theoretically possible* for `quantizer` to be out of bounds, but that's more a bug inside the decoder than here